src/pyspiffe/workloadapi/default_jwt_source.py

```diff
@@ -0,0 +1,149 @@
+    """
+
+    def __init__(
+        self,
+        audiences: Set[str],
+        workload_api_client: WorkloadApiClient = None,
+        spiffe_socket_path: str = None,
+        timeout_in_seconds: float = None,
+        subject: Optional[SpiffeId] = None,
+    ) -> None:
+        """Creates a new DefaultJwtSource.
+           It blocks until the initial update has been received from the Workload API or until timeout_in_seconds is reached.
+           In case the underlying Workload API connection returns an unretryable error, the source will be closed and
+           no methods on the source will be available.
+        Args:
+            audiences: List of audiences for the JWT SVID.
+            workload_api_client: A WorkloadApiClient object that will be used to fetch the JWT materials from the Workload API.
+                                 In case it's not provided, a default client will be created.
+            spiffe_socket_path: Path to Workload API UDS. This will be used in case a the workload_api_client is not provided.
+                           If not specified, the SPIFFE_ENDPOINT_SOCKET environment variable will be used and thus, must be set.
+            timeout_in_seconds: Time to wait for the first update of the Workload API. If not provided, and
+                                the connection with the Workload API fails, it will block indefinitely while
+                                the connection is retried.
+            subject: SPIFFE ID subject for the JWT.
+        Returns:
+            DefaultJwtSource: New DefaultJwtSource object, initialized with the JwtBundleSet fetched from the Workload API.
+        Raises:
+            ArgumentError: If spiffe_socket_path is invalid or not provided and SPIFFE_ENDPOINT_SOCKET env variable is not set.
+            JwtSourceError: In case a timeout was configured and it was reached during the source initialization waiting
+                             for the first update from the Workload API.
+        """
+
+        self._initialized = threading.Event()
+        self._lock = threading.Lock()
+        self._closed = False
+        self._audiences = audiences
+        self._subject = subject
+        self._workload_api_client = (
+            workload_api_client
+            if workload_api_client
+            else DefaultWorkloadApiClient(spiffe_socket_path)
+        )
+
+        # set the watcher that will keep the source updated and log the underlying errors
+        self._client_cancel_handler = self._workload_api_client.watch_jwt_bundles(
+            self._set_jwt_data, self._on_error
+        )
+        self._initialized.wait(timeout_in_seconds)
+
+        if not self._initialized.is_set():
+            self._client_cancel_handler.cancel()
+            raise JwtSourceError(
+                'Could not initialize JWT Source: reached timeout waiting for the first update'
+            )
+
+    def get_jwt_svid(self) -> JwtSvid:
```

Comment:
This method should take `audience` and `subject` as parameters and fetch the token, thus `fetch_jwt_svid` would be more appropriate.